app/assets/javascripts/discourse/app/lib/grant-badge-utils.js

```diff
@@ -0,0 +1,30 @@
+import { convertIconClass } from "discourse-common/lib/icon-library";
+
+export function grantableBadges(allBadges, userBadges) {
```

Comment:
I understand that these functions have been extracted from the `mixins/grant-badge-controller`. How easy would it be to update the mixin to lean on these new `lib/` functions? Or even better, remove the mixin entirely and have any consumers use this lib directly?

Main thing is to avoid copy/pasting the code, and then ending up with two almost-identical places to maintain.

Reply:
I'm aligned, I can try doing it in this PR, would lean towards removing the mixin entirely.